### PR TITLE
Assign, equality, equiv

### PR DESCRIPTION
**Pull request by dlecocq**
Add three more methods that `url-py` offers per feedback in https://github.com/seomoz/url-py/pull/41

@b4hand @lindseyreno @tammybailey

### CODE REVIEW

File: test/test-url.cpp [hunk 1]
```diff
@@ -247,6 +247,228 @@ TEST(ParseTest, TestPortNotNumber)
...
+    EXPECT_EQ(a, b);
+    b.setFragment("new-fragment");
+    EXPECT_NE(a, b);
+}
+
+TEST(EquivTest, HttpPort)
+{
+    Url::Url a("http://foo.com:80");
+    Url::Url b("http://foo.com/");
+    EXPECT_EQ(true, a.equiv(b));
```
Comment (neilmb) [inline test/test-url.cpp]: I think `EXPECT_TRUE` is clearer of your intent: https://github.com/google/googletest/blob/master/googletest/docs/Primer.md#basic-assertions here and in the following.

File: test/test-url.cpp [hunk 2]
```diff
@@ -247,6 +247,228 @@ TEST(ParseTest, TestPortNotNumber)
...
+    EXPECT_EQ(true, b.equiv(a));
+    EXPECT_NE(a, b);
+}
+
+TEST(NotEquivTest, HostnameMismatch)
+{
+    Url::Url a("http://foo.com:");
+    Url::Url b("http://foo.co.uk/");
+    EXPECT_EQ(false, a.equiv(b));
```
Comment (neilmb) [inline test/test-url.cpp]: `EXPECT_FALSE`

File: test/test-url.cpp (hunk 1 above)
Comment (dlecocq) [inline test/test-url.cpp]: Thanks -- I was too lazy to look up whether there was such an expect last night, but updated.

File: include/url.h [hunk 3]
```diff
@@ -79,6 +79,22 @@ namespace Url
             , fragment_(other.fragment_)
             , userinfo_(other.userinfo_) { }
 
+        /**
+         * Take on the value of the other URL.
+         */
+        Url& assign(const Url& other);
```
Comment (b4hand) [inline include/url.h:85]: Is there a reason not to make this `operator=`?

Comment (dlecocq) [inline include/url.h:85]: In fact, `assign` delegates out to the default `operator=` below, but `assign` is present to be more accessible from Cython.

Comment (dlecocq) [inline include/url.h:85]: By which I mean, the Cython implementation uses the PIMPL idiom, but we cannot do this:

```
# Can't dereference in the normal way
*self.ptr = *other.ptr

# Instead we have to use 'dereference', but now this is nonsensical:
dereference(self.ptr) = dereference(other.ptr)

# Also, we can't access the `operator=` member on self.ptr, AFAIK
```

**Comment by b4hand** [comment]
LGTM

**Pull request merged**